### DISCUSSION TITLE
Implementing a NN on the AIDeck - Problems with wifi streaming.

### DISCUSSION THREAD

**Discussion by finnfi** [bitcraze/discussions#104] [Q&A] [answered]
Hi! 
I am trying to implement a NN on the AIDeck.

I used the wifi streaming example as a base to get started with communication between the gap8, the stm32 and the host computer. I have noticed that sometimes the wifi stream stops. Sometimes it resumes after some time or after I turn my wifi connection on and off. It stops in the "sendBufferViaCPX" function, where the cpxSendPacketBlocking blocks. Usually when the offset value is 8160, which is weird. What could that be? 

Another question I have is about the communication between the stm32 and gap8. Right now am I using the cpx protocol. The gap8 sends a request to the stm32, which in returns sends the state of the drone (which we want as input the NN). I've added a CPXFunction_t enum at both ends, and modified aideck_router.c to get data from the aideck to my app. Is that the right way to do it? 

My last question is about debugging. Right now I only use the radio to flash and DEBUG_PRINT(stm32)/cpxPrinttoConsole(gap8) to print debug stuff. I am pretty new to multiple mcu debugging. Do you have some advice for what I can do? I have several JTAGs. For example would it be nice with console output from both the stm32 and the gap8. 

Hope you can help me! :)

**Comment by finnfi** [comment, marked as answer]
The wifi issue seems to be my wifi driver :)

**Reply by knmcguire** [reply]
Great!

For the other questions, could you please make a separate discussion thread for each one of them if they are still unanswered for you? We prefer to have one topic/question per discussions. (see https://github.com/orgs/bitcraze/discussions/55)
